src/CPUSolver.cpp

```diff
@@ -2202,44 +2202,50 @@ void CPUSolver::stabilizeFlux() {
 
 
 /**
- * @brief Computes the volume-averaged, energy-integrated nu-fission rate in
- *        each FSR and stores them in an array indexed by FSR ID.
+ * @brief Computes the volume-averaged, energy-integrated fission or nu-fission
+ *        rate in each FSR and stores them in an array indexed by FSR ID.
  * @details This is a helper method for SWIG to allow users to retrieve
- *          FSR nu-fission rates as a NumPy array. An example of how this
- *          method can be called from Python is as follows:
+ *          FSR fission or nu-fission rates as a NumPy array. An example of
+ *          how this method can be called from Python is as follows:
  *
  * @code
  *          num_FSRs = geometry.getNumFSRs()
  *          fission_rates = solver.computeFSRFissionRates(num_FSRs)
  * @endcode
  *
- * @param fission_rates an array to store the nu-fission rates (implicitly
+ * @param fission_rates an array to store the fission rates (implicitly
  *                      passed in as a NumPy array from Python)
  * @param num_FSRs the number of FSRs passed in from Python
+ * @param nu whether to use nu-fission rates instead of fission rates
  */
-void CPUSolver::computeFSRFissionRates(double* fission_rates, long num_FSRs) {
+void CPUSolver::computeFSRFissionRates(double* fission_rates, long num_FSRs, bool nu = false) {
 
   if (_scalar_flux == NULL)
     log_printf(ERROR, "Unable to compute FSR fission rates since the "
                "source distribution has not been calculated");
 
   log_printf(INFO, "Computing FSR fission rates...");
 
-  FP_PRECISION* nu_sigma_f;
+  FP_PRECISION* sigma_f;
   FP_PRECISION vol;
 
   /* Initialize fission rates to zero */
   for (long r=0; r < _num_FSRs; r++)
     fission_rates[r] = 0.0;
 
   /* Loop over all FSRs and compute the volume-weighted fission rate */
-#pragma omp parallel for private (nu_sigma_f, vol) schedule(static)
+#pragma omp parallel for private (sigma_f, vol) schedule(static)
   for (long r=0; r < _num_FSRs; r++) {
-    nu_sigma_f = _FSR_materials[r]->getNuSigmaF();
+    /* TODO: move the 'if' block outside */
```

Comment:
@GiudGiud, I don't have much experience with C++. Is it possible to move the `if-else` block before the `for` loop and select the getSigmaF() function there?

Reply:
It's fine like that. The penalty for an if statement here is very small